sqlglot/parser.py

```diff
@@ -1551,10 +1574,55 @@ def extend_props(temp_props: t.Optional[exp.Properties]) -> None:
             begin=begin,
             end=end,
             clone=clone,
-            start=seq_start,
-            increment=seq_increment,
         )
 
+    def _parse_create_sequence(self, replace, properties) -> exp.Create | exp.Command:
+        exists = self._parse_exists(not_=True)
+        this = self._parse_table(schema=True)
+
+        seq = exp.Create(
+            this=this,
+            exists=exists,
+            replace=replace,
+            kind="SEQUENCE",
+            properties=properties,
+        )
+
+        options = []
+
+        while self._curr:
+            if self._match_text_seq("AS"):
+                seq.set("data_type", self._parse_types())
+            elif self._match_text_seq("INCREMENT"):
+                self._match_text_seq("BY")
+                seq.set("increment", self._parse_term())
+            elif self._match_text_seq("MINVALUE"):
+                seq.set("minvalue", self._parse_term())
+            elif self._match_text_seq("MAXVALUE"):
+                seq.set("maxvalue", self._parse_term())
+            elif self._match(TokenType.START_WITH) or self._match_text_seq("START"):
+                seq.set("start", self._parse_term())
```

Comment:
Don't you need `_parse_equality` here (and possibly elsewhere) to consume the `EQ`? From Snowflake's docs:

```
[ START [ WITH ] [ = ] <initial_value> ]
```

Reply:
Good catch, Snowflake does allow for an optional `EQ`. As a general note which can apply in future PRs too, since other dialects do not seem to specify `EQ` in their syntax and it's optional anyways, would it be better to consume the `EQ` token (just for Snowflake) and `_parse_term()` - as a way of canonicalizing `CREATE SEQUENCE` while also simplifying transpilation - or preserve the full roundtrip by going with `_parse_equality()` instead?